resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesTaskSchedulerImplSuite.scala

```diff
@@ -0,0 +1,142 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.scheduler.cluster.kubernetes
+
+import io.fabric8.kubernetes.api.model.{Pod, PodSpec, PodStatus}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.CommonConfigurationKeysPublic
+import org.mockito.Matchers._
+import org.mockito.Mockito._
+
+import org.apache.spark.{SparkContext, SparkFunSuite}
+import org.apache.spark.scheduler.FakeTask
+
+class KubernetesTaskSchedulerImplSuite extends SparkFunSuite {
+
+  SparkContext.clearActiveContext()
+  val sc = new SparkContext("local", "test")
+  val backend = mock(classOf[KubernetesClusterSchedulerBackend])
+
+  test("Create a k8s task set manager") {
+    val sched = new KubernetesTaskSchedulerImpl(sc)
+    sched.kubernetesSchedulerBackend = backend
+    val taskSet = FakeTask.createTaskSet(0)
+
+    val manager = sched.createTaskSetManager(taskSet, maxTaskFailures = 3)
+    assert(manager.isInstanceOf[KubernetesTaskSetManager])
+  }
+
+  test("Gets racks for datanodes") {
+    val rackResolverUtil = mock(classOf[RackResolverUtil])
+    when(rackResolverUtil.isConfigured).thenReturn(true)
+    when(rackResolverUtil.resolveRack(sc.hadoopConfiguration, "kube-node1"))
+      .thenReturn(Option("/rack1"))
+    when(rackResolverUtil.resolveRack(sc.hadoopConfiguration, "kube-node2"))
+      .thenReturn(Option("/rack2"))
+    val sched = new KubernetesTaskSchedulerImpl(sc, rackResolverUtil)
+    sched.kubernetesSchedulerBackend = backend
+    when(backend.getExecutorPodByIP("kube-node1")).thenReturn(None)
+    when(backend.getExecutorPodByIP("kube-node2")).thenReturn(None)
+
+    assert(sched.getRackForHost("kube-node1:60010") == Option("/rack1"))
+    assert(sched.getRackForHost("kube-node2:60010") == Option("/rack2"))
+  }
+
+  test("Gets racks for executor pods") {
+    val rackResolverUtil = mock(classOf[RackResolverUtil])
+    when(rackResolverUtil.isConfigured).thenReturn(true)
+    when(rackResolverUtil.resolveRack(sc.hadoopConfiguration, "kube-node1"))
+      .thenReturn(Option("/rack1"))
+    when(rackResolverUtil.resolveRack(sc.hadoopConfiguration, "kube-node2.mydomain.com"))
+      .thenReturn(Option("/rack2"))
+    when(rackResolverUtil.resolveRack(sc.hadoopConfiguration, "kube-node2"))
+      .thenReturn(None)
+    when(rackResolverUtil.resolveRack(sc.hadoopConfiguration, "192.168.1.5"))
+      .thenReturn(None)
+    val inetAddressUtil = mock(classOf[InetAddressUtil])
```

Comment:
If it can't be then move `InetAddressUtil` to its own file and place a `trait` over it. The real `InetAddressUtil` could probably be an `object` that extends the trait in this case.